sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -425,10 +443,11 @@ class SessionCatalog(
   def tableExists(name: TableIdentifier): Boolean = synchronized {
```

Comment:
can you update SessionCatalogSuite to reflect this behavior? I think we weren't checking temp tables in the past.